### DISCUSSION TITLE
Prefix: Container Status Thoughts

### DISCUSSION THREAD

**Discussion by jvanderaa** [nautobot/nautobot#1309] [Ideas]
It came up recently that it may make more sense for a Prefix that is defined as a container to have a separate boolean check box for this. The status is used more or less as a `type` of prefix instead of a status of the Prefix. By moving to a type, maybe not a boolean, this would allow for a real status such as `Active`/`Decommissioned`/`Maintenance` type thing, while keeping the type of Prefix maintained as well. This would likely be a breaking change though?

**Comment by lampwins** [comment]
I do actually really like this idea. There is merit in the fact that the type should be used for IPAM related business logic while the status should provide organizational level meaning. @jvanderaa let's get an issue open for this, please. That said, I am not sure how we would be able to do this in a non-breaking way so it would likely be a 2.0 feature, but we can investigate this.

**Comment by itdependsnetworks** [comment]
Does it make sense to add to https://github.com/nautobot/nautobot/issues/1032 ?

**Reply by jvanderaa** [reply]
We could look at maybe adding `prefix_type` to the model in 1.3. But migrating the logic for `container` or `pool` type scenarios for displays and logic, that change may be a breaking change there.

**Comment by susanhooks** [comment]
This came up in conversation today around issue #1428 . The end user is using containers for 10.0.0.0/8 space down several levels, as was advised to make more precise adding additional IP addresses over the large space, and to prevent the IP from getting assigned to the 'wrong' prefix by default. The architecture keeps lower prefixes separate and aids navigation, but there's some unexpected behavior around % Utilization. While marked as `container`, the utilization of top level showed 100%, when switched to `active`, it showed as 0%.

**Comment by jvanderaa** [comment]
@lampwins @bryanculver is this prioritized?

**Comment by bryanculver** [comment]
@jvanderaa These seem related, can you confirm: https://github.com/nautobot/nautobot/issues/1969, https://github.com/nautobot/nautobot/issues/1362

**Comment by jvanderaa** [comment]
Thanks @bryanculver 

Indeed that #1362 captures this. I'll close #1969
